### ISSUE TITLE
calcite-shell-panel: Allow chevron to represent button action rather than current state

### ISSUE THREAD

**Issue by tayMAPNeil** [Esri/calcite-design-system#10215]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

In the new arcgis-tabs-manager component, the collapse chevron represents the current state of the panel, rather than the action if the button is selected. 

### Acceptance Criteria

Be able to change the direction of the collapse/expand caret icon to represent the "action" of the button rather than the current "state". 

### Relevant Info

Related to: https://devtopia.esri.com/WebGIS/arcgis-web-analysis/issues/7035

We have other double chevrons in the panle that are used to set height up/height down and they are positioned so that they represent the "action" rather than the current "state". We would like the collapse button icon to be consistent with this. 

### Which Component

calcite-panel / calcite-shell-panel

### Example Use Case

https://codepen.io/khoomgayu/pen/xxoaryN?editors=100

`arcgis-tabs-manager` behaviour: 

https://github.com/user-attachments/assets/12c532e5-1d90-42bb-9cd9-b60def21324b

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online

**Comment by amring1** [comment]
This got reported as a bug by almost all of the testers in the ModelBuilder Holistic testing event. It would be really great to have this resolved for R3

**Comment by ashetland** [comment]
I'm not sure we need to change the pattern across the system. Panel is unique here because when it's placement is docked to the bottom, I'd expect that chevron to be inverted (same for Shell Panel in the `panel-bottom` slot). When Panel is at the top, left, or right the current direction makes sense. A placement prop seems out of scope for Panel, but maybe something like collapseDirection would be useful in these situations?

**Comment by amring1** [comment]
Agreed that it doesn't necessarily require a system-level change, but a property to be able to specify a collapseDirection makes sense to me and would be a great improvement!

**Comment by ashetland** [comment]
Brainstorming some possibilities:
- `invert-collapse` simple boolean, but a touch ambiguous
- `collapse-direction="up" (default) | "down"` aligns with related prop names and is more descriptive

cc @driskull For additional thoughts on this issue.

**Comment by driskull** [comment]
> I'm not sure we need to change the pattern across the system. Panel is unique here because when it's placement is docked to the bottom, I'd also expect that chevron to be inverted (same for Shell Panel in the panel-bottom slot). When Panel is at the top, left, or right the current chevron direction makes sense. A placement prop seems out of scope for Panel, but maybe something like collapseDirection would be useful in these situations?

@ashetland this will be pretty difficult to explain to end users when to use one vs the other. That's my concern.

Also, panel isn't unique. An accordion, block, etc could be docked to the bottom of a map.

My concern is that this property is going to create a lot of inconsistencies throughout components and applications.

If we do go this route, I prefer `collapase-direction`.

**Comment by ashetland** [comment]
> this will be pretty difficult to explain to end users when to use one vs the other

My concern as well. To play devil's advocate, is this any different from other docs situations?

**Comment by driskull** [comment]
>  is this any different from other docs situations?

Probably not 😃 

At least the direction option is clearer.

**Comment by macandcheese** [comment]
> > Ideally that resize behavior is achieved using ‘resizable’ functionality - not with custom UI.
> 
> With a built-in resizing affordance on the top edge of Panel in this situation, would the current chevron direction make sense or would it still feel like it's pointing in the wrong direction?

That use case is "extra noticeable" because there is the direct opposite visual / action right next to the chevron. Some systems use "plus / minus" for this reason (it's an Accordion option for us) .. but that has it's own set of issues, and potential conflation with "add or remove" actions ... 

`collapse-direction` seems to make sense from a naming standpoint, though I think the implementation will end up being a bit varied.

**Comment by driskull** [comment]
We are using this internal property in the Maps SDK. Should we bump this one up to document the `collapseDirection` property? cc @geospatialem @jcfranco

**Comment by DitwanP** [comment]
Bumping Matt's comment above @jcfranco @geospatialem

**Comment by isaacbraun** [comment]
@driskull I guess it wasn't clear to me what was decided as the description wasn't updated, although I guess I did miss `shell-panel. `flow-item` already has it implemented (needs to be exposed), but do we want to add it to specific components or all that have expand/collapse UI?

cc @ashetland @geospatialem

**Comment by ashetland** [comment]
I think we can limit the scope of this to Panel for the time being.
